File: 品味Java的21種設計模式/dp_cpp/factorymethod/example3/ExportDBOperate.cpp
#include "ExportDBOperate.h"

namespace cn {
namespace javass {
namespace dp {
namespace factorymethod {
namespace example3 {

ExportFileApi *ExportDBOperate::factoryMethod() {
    //创建导出成数据库备份文件形式的对象
    return new ExportDB();
}
}
}
}
}
}

File: 品味Java的21種設計模式/dp_java/factorymethod/example3/ExportDBOperate.java
package cn.javass.dp.factorymethod.example3;
/**
 * 具体的创建器实现对象，实现创建导出成数据库备份文件形式的对象
 */
public class ExportDBOperate extends ExportOperate {
    protected ExportFileApi factoryMethod() {
        //创建导出成数据库备份文件形式的对象
        return new ExportDB();
    }
}
